Cython/Compiler/UtilityCode.py

```diff
@@ -265,6 +267,60 @@ def filter_inherited_directives(current_directives):
         return utility_code_directives
 
 
+class CythonSharedUtilityCode:
+    def __init__(self, module_name, context, requires):
+        self._module_name = module_name
+        self.context = context
+        self.requires = requires
+        self.pxd_scope = None
+
+    def find_module(self, context):
+        import Cython
+        from .Scanning import FileSourceDescriptor
+        from .Errors import CompileError
```

Comment:
This (again) suggests to me that at least some of this code lives in the wrong place.